diff --git a/src/auth/auth.service.ts b/src/auth/auth.service.ts
index b7918f2..8a10533 100644
--- a/src/auth/auth.service.ts
+++ b/src/auth/auth.service.ts
@@ -59,7 +59,7 @@ export class AuthService {
     let user = await this.prisma.user.findFirst({
       where: {
         provider: oauthUser.provider,
-        providerId: oauthUser.providerId,
+        sub: oauthUser.providerId,
       },
     });
 
@@ -68,7 +68,7 @@ export class AuthService {
         data: {
           email: oauthUser.email,
           provider: oauthUser.provider,
-          providerId: oauthUser.providerId,
+          sub: oauthUser.providerId,
         },
       });
     }
diff --git a/src/auth/google.strategy.ts b/src/auth/google.strategy.ts
index 95b7b70..a55ea50 100644
--- a/src/auth/google.strategy.ts
+++ b/src/auth/google.strategy.ts
@@ -6,8 +6,8 @@ import { Strategy, Profile } from 'passport-google-oauth20';
 export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
   constructor() {
     super({
-      clientID: process.env.GOOGLE_CLIENT_ID,
-      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+      clientID: process.env.GOOGLE_CLIENT_ID as string,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
       callbackURL: 'http://localhost:3000/auth/google/callback',
       scope: ['email', 'profile'],
     });
@@ -16,7 +16,7 @@ export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
 
   async validate(accessToken: string, refreshToken: string, profile: Profile) {
     return {
-      email: profile.emails[0].value,
+      email: profile.emails?.[0]?.value,
       name: profile.displayName,
     };
   }
diff --git a/src/auth/infoteam.strategy.ts b/src/auth/infoteam.strategy.ts
index ae3c0d4..ed00445 100644
--- a/src/auth/infoteam.strategy.ts
+++ b/src/auth/infoteam.strategy.ts
@@ -9,8 +9,8 @@ export class InfoTeamStrategy extends PassportStrategy(Strategy, 'infoteam') {
     super({
       authorizationURL: '인포팀_AUTH_URL',
       tokenURL: '인포팀_TOKEN_URL',
-      clientID: config.get('INFOTEAM_CLIENT_ID'),
-      clientSecret: config.get('INFOTEAM_CLIENT_SECRET'),
+      clientID: config.get<string>('INFOTEAM_CLIENT_ID') as string,
+      clientSecret: config.get<string>('INFOTEAM_CLIENT_SECRET') as string,
       callbackURL: 'http://localhost:3000/auth/infoteam/callback',
       scope: ['profile'],
     });
diff --git a/src/post/post.controller.ts b/src/post/post.controller.ts
index ee4ae4a..27cf34c 100644
--- a/src/post/post.controller.ts
+++ b/src/post/post.controller.ts
@@ -31,7 +31,7 @@ export class PostController {
   // userId 조회
   @Get('/user')
   getByUser(@Query('userId') userId: string) {
-    return this.service.getPostsByUserId(+userId);
+    return this.service.getPostsByUserId(userId);
   }
 
   // 생성
